### PR TITLE
Fix #9: autogenerate cluster-preflight-check

### LINKED ISSUE #9: Replace bin/cluster-preflight-check with autogenerated script from setup.py
## Problem

You don't need the script in `bin/cluster-preflight-check`. The setuptools allows you to auto generate the script, be it for Python2 or 3.

## Benefits

* You don't need a separate, hand-crafted script
* It's done all automatically through `setup.py`
* It's Python standard packaging practice
* The combination of Python commands and shell script looks potentially insecure to me; we shouldn't reinvent the wheels again if we can avoid that with a standard method. :wink:

Cons:

* You need a virtual environment to generate the script
   (I don't consider it as a real disadvantage as you need a virtual environment anyway for your testing).

## Expected Solution

In a nutshell, pass the `entry_points` parameter to the `setuptools.setup` function:

* Remove the file `bin/cluster-preflight-check`.
* Remove in `setup.py` the following line:

    ```python
    scripts=['bin/cluster-preflight-check'],
    ```

* Add another function `main()` in your `cluster_preflight_check/main.py` file which the following content:

    ```python
    def main():
        ctx = Context()
        run(ctx)
    ```

* Add the following lines into your `setup.py`.

    ```python
    entry_points={
        'console_scripts': ['cluster-preflight-check=cluster_preflight_check.main:main'],
    }
    ```

### PR DESCRIPTION
**Pull request by tomschr**
This PR contains the following changes to fix #9:

* Remove the file `bin/cluster-preflight-check`
* Change `setup.py` to contain the `entry_points` keyword
* Insert missing `main()` function

### CODE REVIEW

File: cluster_preflight_check/main.py [hunk 1]
```diff
@@ -387,4 +387,7 @@ def run(context):
         sys.exit(1)
 
 
-ctx = Context()
+def main():
+    ctx = Context()
```
Comment (liangxin1300) [inline cluster_preflight_check/main.py]: should assign name here:
`ctx.name = os.path.basename(sys.argv[0])`
Thanks!

**Comment by liangxin1300** [comment]
> This PR contains the following changes to fix #9:
> 
> * Remove the file `bin/cluster-preflight-check`
> * Change `setup.py` to contain the `entry_points` keyword
> * Insert missing `main()` function

Cool!

File: cluster_preflight_check/main.py (hunk 1 above)
Comment (tomschr) [reply cluster_preflight_check/main.py]: Yes, that's a good idea. :+1:  I missed that part.

Comment (liangxin1300) [reply cluster_preflight_check/main.py]: hi @tomschr , could you please append that line? Then I can merge this PR:) Thanks

Comment (tomschr) [reply cluster_preflight_check/main.py]: @liangxin1300 Ahh, sorry for the delay. I missed your message.
I've included the line and pushed it into the branch so you can merge it. :wink:

**Comment by liangxin1300** [comment]
Thanks!:)

**Pull request merged**